jaxrs/src/main/java/com/facebook/airlift/jaxrs/JaxrsBinder.java

```diff
@@ -47,4 +56,42 @@ public void bindInstance(Object instance)
     {
         resourceBinder.addBinding().toInstance(instance);
     }
+
+    public void bindWithRolesMapping(Class<?> implementation, Map<String, String> roleMap)
+    {
+        synchronizeExternalRoles(implementation, roleMap);
+        bind(implementation);
+    }
+
+    private void synchronizeExternalRoles(Class<?> implementation, Map<String, String> roleMap)
+    {
+        // Checking class level annotation
+        if (implementation.isAnnotationPresent(RolesAllowed.class)) {
+            processRolesMapping(implementation.getAnnotation(RolesAllowed.class), roleMap);
+        }
+        // Checking method level annotation
+        for (Method method : implementation.getMethods()) {
+            if (method.isAnnotationPresent(RolesAllowed.class)) {
+                processRolesMapping(method.getAnnotation(RolesAllowed.class), roleMap);
+            }
+        }
+    }
+
+    private void processRolesMapping(Annotation annotation, Map<String, String> roleMap)
```

Comment:
I believe I don't understand how it works. Could you please elaborate a little bit?

But honestly I would implement this feature in a separate PR. Let's first merge whatever is already done.

Reply:
I'll revert this commit